### PR TITLE
Use bundler/gem_tasks instead of jeweler

### PR DESCRIPTION
**Pull request by hogelog**
Current master Gemfile and gemspec specify ambiguous dependency version.
It makes gem package building difficulty.
(`bundle install && bundle build` fails in most environments.)

This PR makes

- Remove jeweler
  - jeweler is too complicated compared to bundler/gem_tasks
- Fix dependency versions.

These fix bellow commands.

- [x] bundle exec rake build
- [x] bundle exec rake spec
- [x] bundle exec rake rdoc

@cookpad/dev-infra please review

### CODE REVIEW

File: guard_against_physical_delete.gemspec [hunk 1]
```diff
@@ -21,34 +18,14 @@ Gem::Specification.new do |s|
...
-    s.add_dependency(%q<rspec>, [">= 0"])
-    s.add_dependency(%q<sqlite3>, [">= 0"])
-    s.add_dependency(%q<rdoc>, ["~> 3.12"])
-    s.add_dependency(%q<bundler>, ["~> 1.0.0"])
-    s.add_dependency(%q<jeweler>, ["~> 1.8.3"])
-    s.add_dependency(%q<database_cleaner>, [">= 0"])
-  end
+  s.add_runtime_dependency "activerecord", ">= 3.0.10"
+  s.add_development_dependency "rake", "< 12.0"
+  s.add_development_dependency "rspec", ">= 0", "< 2.99"
```
Comment (eagletmt) [inline guard_against_physical_delete.gemspec]: `>= 0` is redundant

File: Gemfile [hunk 2]
```diff
@@ -1,20 +1,3 @@
 source "http://rubygems.org"
```
Comment (eagletmt) [inline Gemfile]: https

File: guard_against_physical_delete.gemspec (hunk 1 above)
Comment (hogelog) [inline guard_against_physical_delete.gemspec]: Exactly!

File: Gemfile (hunk 2 above)
Comment (hogelog) [inline Gemfile]: :raising_hand_man:

**Pull request merged**